internal/kafka/internal/services/kafka.go

```diff
@@ -886,13 +910,32 @@ func (k *kafkaService) GetManagedKafkaByClusterID(clusterID string) ([]managedka
 		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "unable to list kafka requests")
 	}
 
+	var certificate kafka_tls_certificate_management.Certificate
+
+	enableKafkaExternalCertificate := k.kafkaTLSCertificateManagementService.IsKafkaExternalCertificateEnabled()
+
 	var res []managedkafka.ManagedKafka
 	// convert kafka requests to managed kafka
 	for _, kafkaRequest := range kafkaRequestList {
-		mk, err := buildManagedKafkaCR(kafkaRequest, k.kafkaConfig, k.keycloakService)
+		if enableKafkaExternalCertificate { // only fetch certs when Kafka external certificates is enabled
+			certRequest := kafka_tls_certificate_management.GetCertificateRequest{
+				TLSCertRef: kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef,
+				TLSKeyRef:  kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef,
+			}
+
+			var err error
+			certificate, err = k.kafkaTLSCertificateManagementService.GetCertificate(context.Background(), certRequest)
+			if err != nil {
+				return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to find kafka %q certificates", kafkaRequest.ID)
+			}
+
+		}
+
+		mk, err := buildManagedKafkaCR(kafkaRequest, k.kafkaConfig, k.keycloakService, certificate, enableKafkaExternalCertificate)
```

Comment:
If `enableKafkaExternalCertificate` is false, we reuse the last certificate we got on line 927. Is this on purpose?

Reply:
nvm. ignore this

Reply:
I think you are right. I'll move over the certificate variable inside the loop.

Reply:
Yeah, that was the same that happened to me. Then I noticed that the value of `enableKafkaExternalCertificate` never changes (and thus the issue does not exists). I agree that it would be more readable with a non global variable.